src/components/SearchShopByName.vue

```diff
@@ -0,0 +1,49 @@
+<template>
+    <div class="search-by-name">
+        <input type="text" v-model="query" v-on:input="showSearchResults()">
+        <shop-list :shops="shops" />
+   </div>
+</template>
+
+<script>
+import { requestSearchByName } from '@/io/io';
+import ShopList from '@/components/ShopList';
+
+export default
+{
+    name: 'search-shop-by-name',
+    data ()
+    {
+        return {
+            query: '',
+            shops: [],
```

Comment:
to do: result of found stores needs to be passed to store as other component may / will use it. In this case I see this as important for passing this data in to shop map view to let map component render only results that user is searching for.

Reply:
Could you specify a valid use case for that?
I can't imagine that user would like to see a map with marked shops with name matching eg. "fa".

However that point would be valid with "search by tag" -> "As a userd I want to see all liquor shops in my area"

Reply:
Accepted on this level.